### PR TITLE
Create release r1.1 with network-access-management v0.1.0-rc.1

### PR DESCRIPTION
**Pull request by caubut-charter**
Create release v0.1.0.

#### What type of PR is this?

Add one of the following kinds:
* subproject management

#### What this PR does / why we need it:

Prepares for an initial alpha release.

#### Which issue(s) this PR fixes:

N/A

#### Changelog input

N/A

#### Additional documentation

### CODE REVIEW

File: code/API_definitions/network_access_management.yaml [hunk 1]
```diff
@@ -132,7 +132,7 @@ info:
     may be used.  If an end-user replaces the equipment, all existing isolated networks SHOULD be synced to the new
     device representing this mesh.  Alternatively, a synthetic hardware address may also be used if the specification
     for the address type allows.
-  version: wip
+  version: 0.1.0
```
Comment (RandyLevensalor) [inline code/API_definitions/network_access_management.yaml:135]: Leave this as wip until the commonalities and ICM changes are finalized and we've incorporated them.

File: documentation/API_documentation/network-access-management-API-Readiness-Checklist.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# API Readiness Checklist
+
+Checklist for network-access-management v0.1.0 in r1.1.
+
+| Nr | API release assets  | alpha | release-candidate |  initial<br>public | stable<br> public | Status | Comments |
+|----|----------------------------------------------|:-----:|:-----------------:|:-------:|:------:|:----:|:----:|
+|  1 | API definition                               |   M   |         M         |    Y    |    M   | Y    | [link](../../code/API_definitions/network_access_management.yaml) |
```
Comment (RandyLevensalor) [inline documentation/API_documentation/network-access-management-API-Readiness-Checklist.md:7]:
```suggestion
|  1 | API definition                               |   M   |         M         |    M    |    M   | Y    | [link](../../code/API_definitions/network_access_management.yaml) |
```

Missed one change from the template

**Review by RandyLevensalor — CHANGES_REQUESTED** [review]

**Comment by hdamker** [comment]
@caubut-charter @RandyLevensalor 

In order to review it I would need to understand the purpose of this PR. I can't find any meeting minutes therefore I'm lacking here the context.

I see three options:

1. you want to do a release with the current content of the API specification and the content as currently here in the checklist - that would from my perspective an alpha release, as it is not yet in line with Commonalities 0.4.0(-rc.1)
2. you plan to resolve #19 and #20, update the checklist accordingly, and then create with this PR here the release r1.1 with network-access-management v0.1.0-rc.1 (the release candidate of v0.1.0). 
3. you want to skip any pre-release, including the release candidate which is expected within the release cycle and create directly the public release v0.1.0

The option 2 would be in line with the intention of the M3 milestone in the release cycle for the meta-release Fall24 (cf https://wiki.camaraproject.org/display/CAM/Meta-release+Fall24). That would give you also the option of bug fixes and/or further alignments until M4 (end of August), where you would set the version then to v0.1.0 and create the public release.
With option 3 any necessary fixes until M4 would lead to another version number (either v0.1.1 or v0.2.0, depending if they are breaking or not).

Let's assume option 2.:
* The title of the PR should be "Create release r1.1 with network-access-management v0.1.0-rc.1"
* The changelog should reflect that it is a pre-release with the (first) release candidate of v0.1.0
* The PR should change within the YAML the version field from "wip" to "0.1.0-rc.1" and the path to .../network-access-management/v0.1rc1 (as this PR will be last to be merged before the actual creation of the release in GitHub)

Further minor comments I can provide if the purpose of the PR is clarified.

File: code/API_definitions/network_access_management.yaml (hunk 1 above)
Comment (hdamker) [reply code/API_definitions/network_access_management.yaml]: > Leave this as wip until the commonalities and ICM changes are finalized and we've incorporated them.

Actually it is correct that the "release PR" will be the one setting the version which will be released. The important point is that all other PRs (e.g. the once addressing #18 and #19) must be merged before the "release PR".
BTW: the PR to create the release should change if possible only the version field and the base path, but not be used to address issues etc.

**Comment by RandyLevensalor** [comment]
> 2. you plan to resolve [Commonalities pass for corrections #19](https://github.com/camaraproject/NetworkAccessManagement/issues/19) and [ ICM pass for corrections #20](https://github.com/camaraproject/NetworkAccessManagement/issues/20), update the checklist accordingly, and then create with this PR here the release r1.1 with network-access-management v0.1.0-rc.1 (the release candidate of v0.1.0).
> 3. you want to skip any pre-release, including the release candidate which is expected within the release cycle and create directly the public release v0.1.0
> 
> The option 2 would be in line with the intention of the M3 milestone in the release cycle for the meta-release Fall24 (cf https://wiki.camaraproject.org/display/CAM/Meta-release+Fall24). That would give you also the option of bug fixes and/or further alignments until M4 (end of August), where you would set the version then to v0.1.0 and create the public release. With option 3 any necessary fixes until M4 would lead to another version number (either v0.1.1 or v0.2.0, depending if they are breaking or not).
> 
> Let's assume option 2.:
> 
> * The title of the PR should be "Create release r1.1 with network-access-management v0.1.0-rc.1"
> * The changelog should reflect that it is a pre-release with the (first) release candidate of v0.1.0
> * The PR should change within the YAML the version field from "wip" to "0.1.0-rc.1" and the path to .../network-access-management/v0.1rc1 (as this PR will be last to be merged before the actual creation of the release in GitHub)
> 
> Further minor comments I can provide if the purpose of the PR is clarified.

@hdamker Your assumption regarding option 2 is corrected.  Thanks for taking the time to review and provide guidance / clarification.

We are working to resolved  [Commonalities pass for corrections #19](https://github.com/camaraproject/NetworkAccessManagement/issues/19) and [ ICM pass for corrections #20](https://github.com/camaraproject/NetworkAccessManagement/issues/20).  These are primarily updates from Commonalities and ICM that have been made since we last updated the API spec.

If I understand you feedback above, we should address issues #19 and #20, then make the following changes to this PR:
- [ ] Rebase if needed
- [ ] Update the PR title "Create release r1.1 with network-access-management v0.1.0-rc.1"
- [ ] Update the changelog to reflect that it is a pre-release with the (first) release candidate of v0.1.0
- [ ] Update version field from "wip" to "0.1.0-rc.1" and the path to .../network-access-management/v0.1rc1 (as this PR will be last to be merged before the actual creation of the release in GitHub)

**Comment by hdamker** [comment]
@caubut-charter @RandyLevensalor is there any update regarding the release preparation? Especially about progress on #19? I'm asking as the "cut-off" date for approved M3 release PRs is tomorrow (August 15th, cf https://wiki.camaraproject.org/display/CAM/2024-08-01+TSC+Minutes). 

Another option is to take the time to properly do the alignment (maybe also the split as proposed in #26) and make the first initial release independent of the Fall24 release, aiming for the participation within Spring25 with one of the next versions.

Just let me know how you want to proceed.

Note: I took the freedom to edit the PR title and to tick off the point in the list above

**Comment by caubut-charter** [comment]
@hdamker Pushing to Spring25 for the meta release is the best option at this point.

I've had some priorities come up that likely won't slow down until after September with releases, conferences, and demos.

**Comment by caubut-charter** [comment]
Closing to redo as non-meta.

**Pull request closed without merging**